### ISSUE TITLE
Finally! A simple string split in C++!

### ISSUE THREAD

**Issue by lefticus** [lefticus/cpp_weekly#43]

**Comment by lefticus** [comment]
https://compiler-explorer.com/z/1Tzac7vK3

**Comment by lefticus** [comment]
Ep340

**Comment by bearbones** [comment]
Note: as of fmt 9.0.0, you have to convert the subrange with `fmt::join(string, "")` as shown: https://godbolt.org/z/aE5jx3653

Otherwise you get the following error:
<details> <summary>Error message</summary>

```
In file included from /opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/format.h:48,
                 from <source>:2:
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h: In instantiation of 'constexpr fmt::v9::detail::value<Context> fmt::v9::detail::make_value(T&&) [with Context = fmt::v9::basic_format_context<fmt::v9::appender, char>; T = const std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>&]':
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1753:29:   required from 'constexpr fmt::v9::detail::value<Context> fmt::v9::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v9::basic_format_context<fmt::v9::appender, char>; type <anonymous> = fmt::v9::detail::type::custom_type; T = const std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]'
 1753 |   return make_value<Context>(val);
      |          ~~~~~~~~~~~~~~~~~~~^~~~~
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1877:77:   required from 'constexpr fmt::v9::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {const std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>&}; Context = fmt::v9::basic_format_context<fmt::v9::appender, char>; Args = {std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>}]'
 1875 |         data_{detail::make_arg<
      |               ~~~~~~~~~~~~~~~~~                                              
 1876 |             is_packed, Context,
      |             ~~~~~~~~~~~~~~~~~~~                                              
 1877 |             detail::mapped_type_constant<remove_cvref_t<T>, Context>::value>(
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 1878 |             std::forward<T>(args))...} {
      |             ~~~~~~~~~~~~~~~~~~~~~~                                           
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1894:38:   required from 'void fmt::v9::print(format_string<T ...>, T&& ...) [with T = {const std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>&}; format_string<T ...> = basic_format_string<char, const std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>&>]'
 1894 |   return {std::forward<Args>(args)...};
      |                                      ^
<source>:10:19:   required from here
   10 |         fmt::print("{}\n", string);
      |         ~~~~~~~~~~^~~~~~~~~~~~~~~~
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:3251:44:   in 'constexpr' expansion of 'fmt::v9::make_format_args<>((* & args#0))'
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1733:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1733 |       formattable,
      |       ^~~~~~~~~~~
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1733:7: note: 'formattable' evaluates to false
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h: In instantiation of 'constexpr decltype (ctx.begin()) fmt::v9::detail::parse_format_specs(ParseContext&) [with T = std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>; ParseContext = compile_parse_context<char, error_handler>; decltype (ctx.begin()) = const char*]':
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:2923:22:   required from 'constexpr fmt::v9::detail::format_string_checker<Char, ErrorHandler, Args>::format_string_checker(fmt::v9::basic_string_view<Char>, ErrorHandler) [with Char = char; ErrorHandler = fmt::v9::detail::error_handler; Args = {std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>}]'
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1733:7: note:  2923 |         parse_funcs_{&parse_format_specs<Args, parse_context_type>...},
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1733:7: note:       |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:3116:47:   required from here
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1733:7: note:  3116 |       detail::parse_format_string<true>(str_, checker(s, {}));
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1733:7: note:       |                                               ^~~~~~~~~~~~~~
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:2713:12: error: use of deleted function 'fmt::v9::detail::fallback_formatter<T, Char, Enable>::fallback_formatter() [with T = std::ranges::subrange<const char*, const char*, std::ranges::subrange_kind::sized>; Char = char; Enable = void]'
 2713 |   auto f = conditional_t<has_formatter<mapped_type, context>::value,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2714 |                          formatter<mapped_type, char_type>,
      |                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2715 |                          fallback_formatter<stripped_type, char_type>>();
      |                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:1105:3: note: declared here
 1105 |   fallback_formatter() = delete;
      |   ^~~~~~~~~~~~~~~~~~
/opt/compiler-explorer/libs/fmt/9.0.0/include/fmt/core.h:2713:12: note: use '-fdiagnostics-all-candidates' to display considered candidates
 2713 |   auto f = conditional_t<has_formatter<mapped_type, context>::value,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2714 |                          formatter<mapped_type, char_type>,
      |                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2715 |                          fallback_formatter<stripped_type, char_type>>();
      |                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
